Add hpa test

totallyGreg opened the pull request:
This adds a test for making sure that a hpa name-<revision> has a spec.targetScaleRef.name that points to the same name -<revision> deployment.

This is the first step in trying to figure out where the discrepancy noticed in https://github.com/istio/istio/issues/28107 is coming from. 


[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure


Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

totallyGreg committed:
First pass at adding hpatest

totallyGreg committed:
Added scaleTargetRef Test for Revision

istio-testing commented:
Hi @totallyGreg. Thanks for your PR.

I'm waiting for a [istio](https://github.com/orgs/istio/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/istio/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=istio%2Fistio).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

howardjohn commented:
/ok-to-test

istio-testing commented:
@totallyGreg: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323710414399213568) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323710414399213568) | `/test lint_istio`
gencheck_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323710415246462976) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323710414399213568) | `/test lint_istio`
gencheck_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323710415246462976) | `/test gencheck_istio`
unit-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323710410230075392) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323710414399213568) | `/test lint_istio`
gencheck_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323710415246462976) | `/test gencheck_istio`
unit-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323710410230075392) | `/test unit-tests_istio`
integ-pilot-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-pilot-k8s-tests_istio/1323710410272018432) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323710414399213568) | `/test lint_istio`
gencheck_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323710415246462976) | `/test gencheck_istio`
unit-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323710410230075392) | `/test unit-tests_istio`
integ-pilot-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-pilot-k8s-tests_istio/1323710410272018432) | `/test integ-pilot-k8s-tests_istio`
integ-security-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-security-k8s-tests_istio/1323710410301378560) | `/test integ-security-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

totallyGreg committed:
nameRev not revName...

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323710414399213568) | `/test lint_istio`
gencheck_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323710415246462976) | `/test gencheck_istio`
integ-pilot-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-pilot-k8s-tests_istio/1323710410272018432) | `/test integ-pilot-k8s-tests_istio`
integ-security-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-security-k8s-tests_istio/1323710410301378560) | `/test integ-security-k8s-tests_istio`
unit-tests_istio | 07b1b319b93e5224c93728af0a8ca6280b9df188 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323733594144772096) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323710415246462976) | `/test gencheck_istio`
integ-pilot-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-pilot-k8s-tests_istio/1323710410272018432) | `/test integ-pilot-k8s-tests_istio`
integ-security-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-security-k8s-tests_istio/1323710410301378560) | `/test integ-security-k8s-tests_istio`
unit-tests_istio | 07b1b319b93e5224c93728af0a8ca6280b9df188 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323733594144772096) | `/test unit-tests_istio`
lint_istio | 07b1b319b93e5224c93728af0a8ca6280b9df188 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323733598326493184) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-pilot-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-pilot-k8s-tests_istio/1323710410272018432) | `/test integ-pilot-k8s-tests_istio`
integ-security-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-security-k8s-tests_istio/1323710410301378560) | `/test integ-security-k8s-tests_istio`
unit-tests_istio | 07b1b319b93e5224c93728af0a8ca6280b9df188 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323733594144772096) | `/test unit-tests_istio`
lint_istio | 07b1b319b93e5224c93728af0a8ca6280b9df188 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323733598326493184) | `/test lint_istio`
gencheck_istio | 07b1b319b93e5224c93728af0a8ca6280b9df188 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323733599156965376) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

totallyGreg committed:
namesapce replaced with namespace

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-security-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-security-k8s-tests_istio/1323710410301378560) | `/test integ-security-k8s-tests_istio`
unit-tests_istio | 07b1b319b93e5224c93728af0a8ca6280b9df188 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323733594144772096) | `/test unit-tests_istio`
lint_istio | 07b1b319b93e5224c93728af0a8ca6280b9df188 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323733598326493184) | `/test lint_istio`
gencheck_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323737226382675968) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-security-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-security-k8s-tests_istio/1323710410301378560) | `/test integ-security-k8s-tests_istio`
unit-tests_istio | 07b1b319b93e5224c93728af0a8ca6280b9df188 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323733594144772096) | `/test unit-tests_istio`
gencheck_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323737226382675968) | `/test gencheck_istio`
lint_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323737225548009472) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-security-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-security-k8s-tests_istio/1323710410301378560) | `/test integ-security-k8s-tests_istio`
gencheck_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323737226382675968) | `/test gencheck_istio`
lint_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323737225548009472) | `/test lint_istio`
unit-tests_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323737218707099648) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-security-k8s-tests_istio | 68ca90034e210e5b1cea052e4cd1b7b15ebea57b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-security-k8s-tests_istio/1323710410301378560) | `/test integ-security-k8s-tests_istio`
gencheck_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323737226382675968) | `/test gencheck_istio`
lint_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323737225548009472) | `/test lint_istio`
unit-tests_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323737218707099648) | `/test unit-tests_istio`
integ-pilot-k8s-tests_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-pilot-k8s-tests_istio/1323737218757431296) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

shamsher31 reviewed (approved):
Thanks.

shamsher31 commented:
/retest

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323737225548009472) | `/test lint_istio`
unit-tests_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323737218707099648) | `/test unit-tests_istio`
integ-pilot-k8s-tests_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-pilot-k8s-tests_istio/1323737218757431296) | `/test integ-pilot-k8s-tests_istio`
gencheck_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323869908169134080) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323737218707099648) | `/test unit-tests_istio`
integ-pilot-k8s-tests_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-pilot-k8s-tests_istio/1323737218757431296) | `/test integ-pilot-k8s-tests_istio`
gencheck_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323869908169134080) | `/test gencheck_istio`
lint_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323869908143968256) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-pilot-k8s-tests_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-pilot-k8s-tests_istio/1323737218757431296) | `/test integ-pilot-k8s-tests_istio`
gencheck_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323869908169134080) | `/test gencheck_istio`
lint_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323869908143968256) | `/test lint_istio`
unit-tests_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323869908022333440) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

shamsher31 commented on operator/cmd/mesh/manifest-generate_test.go:
```diff
@@ -566,13 +566,15 @@ func TestConfigSelectors(t *testing.T) {
 		// First we fetch all the objects for our default install
 		cname := "istiod"
 		deployment := mustFindObject(t, objs, cname, name.DeploymentStr)
+		hpa := mustFindObject(t, objs, cname, name.HPAStr)
```
I think you can remove this line as it's not being used anywhere.

totallyGreg replied:
I'm not sure I understand the overall testing strategy, as the "cross revision" run seems to want to set both a default and a canary and then verify things are not incorrect between them.  The hpa test is already being done in the validateRefererntialIntegrity function but only checking against cname.  I'll remove the extra hpa.

totallyGreg replied:
/retest

totallyGreg committed:
Removed unused hpa definition

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323869908169134080) | `/test gencheck_istio`
lint_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323869908143968256) | `/test lint_istio`
unit-tests_istio | 84eede53ad4e3af2225c9f80481e6a03a38ddfed | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323994317294931968) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 6c2ab97f724211d66bf537ac13c7624fdc0aec95 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/lint_istio/1323869908143968256) | `/test lint_istio`
unit-tests_istio | 84eede53ad4e3af2225c9f80481e6a03a38ddfed | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323994317294931968) | `/test unit-tests_istio`
gencheck_istio | 84eede53ad4e3af2225c9f80481e6a03a38ddfed | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323994322302930944) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 84eede53ad4e3af2225c9f80481e6a03a38ddfed | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323994317294931968) | `/test unit-tests_istio`
gencheck_istio | 84eede53ad4e3af2225c9f80481e6a03a38ddfed | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323995840104108032) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | 84eede53ad4e3af2225c9f80481e6a03a38ddfed | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323995840104108032) | `/test gencheck_istio`
unit-tests_istio | 84eede53ad4e3af2225c9f80481e6a03a38ddfed | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323995840007639040) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

shamsher31 commented:
/retest

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 84eede53ad4e3af2225c9f80481e6a03a38ddfed | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323995840007639040) | `/test unit-tests_istio`
gencheck_istio | 84eede53ad4e3af2225c9f80481e6a03a38ddfed | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323999440431747072) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@totallyGreg: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | 84eede53ad4e3af2225c9f80481e6a03a38ddfed | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/gencheck_istio/1323999440431747072) | `/test gencheck_istio`
unit-tests_istio | 84eede53ad4e3af2225c9f80481e6a03a38ddfed | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/unit-tests_istio/1323999440335278080) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

shamsher31 commented:
@totallyGreg can you run `make gen` and `make refresh-goldens` and add any file changed?

totallyGreg commented:
> @totallyGreg can you run `make gen` and `make refresh-goldens` and add any file changed?

I ran `make gen` and after a lot of fan noise it failed trying to execute the refresh-golden target itself. 
 
```makefile
FAIL
Makefile.core.mk:334: recipe for target 'refresh-goldens' failed
make[1]: *** [refresh-goldens] Error 1
make: *** [gen] Error 2
``` 
The only changed files are go.mod & go.sum.  
I can push these if needed.  

```shell
❯ git status
On branch add-hpa-test
Your branch is up to date with 'origin/add-hpa-test'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   go.mod
        modified:   go.sum
```

shamsher31 commented on operator/cmd/mesh/manifest-generate_test.go:
```diff
@@ -573,6 +573,7 @@ func TestConfigSelectors(t *testing.T) {
 		// Next we fetch all the objects for a revision install
 		nameRev := "istiod-canary"
 		deploymentRev := mustFindObject(t, objsRev, nameRev, name.DeploymentStr)
+		hpaRev := mustFindObject(t, objs, nameRev, name.HPAStr)
```
@totallyGreg  After spending an hour trying to debug a test, found out that you were looking into the wrong object for the Rev label. I ran it locally and now  `make gen` and `make refresh-goldens` are passing.
```suggestion
		hpaRev := mustFindObject(t, objsRev, nameRev, name.HPAStr)
```

shamsher31 commented:
Not sure why the Docs team is asked to review this PR. @ericvn can you PTAL?

totallyGreg commented:
> Not sure why the Docs team is asked to review this PR. @ericvn can you PTAL?

Probably because I was dumb and corrected a series of 3 unrelated typos in some comments.

totallyGreg committed:
objsRev instead of objs

shamsher31 commented:
@totallyGreg can you rebase your PR? I don't think `go.mod` and `go.sum` changes are needed in this PR.

totallyGreg committed:
revert go.sum go.mod

istio-testing commented:
@totallyGreg: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-ipv6-k8s-tests_istio | a872fab8ba86ab0b454c7dacd60aee71f524ba24 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28544/integ-ipv6-k8s-tests_istio/1324421563662995456) | `/test integ-ipv6-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

shamsher31 commented:
/test integ-ipv6-k8s-tests_istio

shamsher31 commented:
cc: @istio/wg-docs-maintainers
